tests/containers/bci_tests.pm

```diff
@@ -0,0 +1,135 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: bci-tests runner
+#   SUSE Linux Enterprise Base Container Images (SLE BCI)
+#   provides truly open, flexible and secure container images and application
+#   development tools for immediate use by developers and integrators without
+#   the lock-in imposed by alternative offerings.
+#   - install needed python environment
+#   - clone bci-tests repo
+#   - build project
+#   - run test for each given environment
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use XML::LibXML;
+use utils qw(systemctl zypper_call ensure_ca_certificates_suse_installed);
+use version_utils qw(get_os_release);
+use containers::common;
+use testapi;
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    my $runtime        = get_required_var('RUNTIME');
+    my $bci_tests_repo = get_required_var('BCI_TESTS_REPO');
+    my $test_envs      = get_required_var('BCI_TEST_ENVS');
+    my $bci_devel_repo = get_var('BCI_DEVEL_REPO');
+    my $bci_timeout    = get_var('BCI_TIMEOUT', 900);
+
+    ensure_ca_certificates_suse_installed;
+
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    if ($runtime eq 'podman') {
+        install_podman_when_needed($host_distri);
+        install_buildah_when_needed($host_distri);
+    }
+    elsif ($runtime eq 'docker') {
+        install_docker_when_needed($host_distri);
+    }
+    else {
+        die("Runtime $runtime not given or not supported");
+    }
+
+    record_info('Install', 'Install needed packages');
+    zypper_call('--quiet in git-core python39', timeout => 600);
+    assert_script_run('pip3.9 --quiet install tox pytest', timeout => 600);
+
+    record_info('Clone', 'Clone BCI tests repository');
+    assert_script_run("git clone -q --depth 1 $bci_tests_repo");
+
+    record_info('Build', 'Build bci-tests project');
+    assert_script_run('cd bci-tests');
+    assert_script_run("export CONTAINER_RUNTIME=$runtime");
+    assert_script_run("export BCI_DEVEL_REPO=$bci_devel_repo") if $bci_devel_repo;
+    assert_script_run('tox -e build', timeout => 600);
+    upload_logs('junit_build_serial.xml');
+    upload_logs('junit_build_parallel.xml');
```

Comment:
Is it necessary to have this part outside of test case loop? I mean, it feels like build tests should be always as the first element of `BCI_TEST_ENVS` list.

Reply:
`build` is a pre-step you need to do before running any tests, it basically sets up the needed packages used by the tests. I would like to keep it separate as it's not an actual test. Maybe @dcermak can comment on this?